### PR TITLE
feature/issue-91

### LINKED ISSUE #91: Use jest-sonar-reporter fork
Use this package: https://github.com/CasualBot/jest-sonar-reporter

RnD research here: #82

### PR DESCRIPTION
**Pull request by ryabukhinstep**

### CODE REVIEW

File: packages/app/package.json [hunk 1]
```diff
@@ -45,6 +45,9 @@
       "ts",
       "js"
     ],
+    "reporters": [
+      "@casualbot/jest-sonar-reporter"
```
Comment (KillWolfVlad) [inline packages/app/package.json]: Return `default` reporter, see https://github.com/CasualBot/jest-sonar-reporter#usage for more usage information.

File: packages/app/package.json [hunk 2]
```diff
@@ -119,9 +121,8 @@
   "engines": {
     "node": ">=16"
   },
-  "jestSonar": {
-    "reportPath": "./coverage",
-    "reportFile": "test-report.xml",
-    "indent": 2
+  "@casualbot/jest-sonar-reporter": {
```
Comment (KillWolfVlad) [inline packages/app/package.json]: Configure reporter via jest, see https://github.com/CasualBot/jest-sonar-reporter#configuration for more usage information.

**Review by KillWolfVlad — CHANGES_REQUESTED** [review]

File: packages/app/package.json (hunk 2 above)
Comment (ryabukhinstep) [reply packages/app/package.json]: What parameters do you think are missing?
I think the given options are enough.

Comment (KillWolfVlad) [reply packages/app/package.json]: See to the bottom of the README:

```javascript
// jest.config.js
{
    reporters: [
        'default', 
        [ 
            '@casualbot/jest-sonar-reporter',
            {
                relativePaths: true,
                outputName: 'sonar-report.xml',
                outputDirectory: 'coverage'
            }
        ]
    ],
}
```

Comment (ryabukhinstep) [reply packages/app/package.json]: Do you think your solution is more readable or more correct?
I used a different configuration option. it is working.

it's just above yours

```
"@casualbot/jest-sonar-reporter": {
    "suiteName": "jest tests",
    "outputDirectory": "coverage",
    "outputName": "jest-report.xml",
    "uniqueOutputName": "false",
    "classNameTemplate": "{classname}-{title}",
    "titleTemplate": "{classname}-{title}",
    "ancestorSeparator": " › ",
    "usePathForSuiteName": "true",
    "relativePaths": "true"
  }
```

Comment (KillWolfVlad) [reply packages/app/package.json]: I want use configuration via Jest to place all thing as more closer as possible. See how we setup Allure reporter:

```json
{
  "jest": {
    "moduleFileExtensions": [
      "ts",
      "js"
    ],
    "resetMocks": true,
    "rootDir": "./",
    "roots": [
      "<rootDir>/src",
      "<rootDir>/test"
    ],
    "setupFilesAfterEnv": [
      "jest-extended/all",
      "./src/main.ts"
    ],
    "testEnvironment": "jest-allure-circus",
    "testEnvironmentOptions": {
      "jiraUrl": "https://jira.example.com/browse/"
    },
    "testRegex": ".*\\.spec\\.ts$",
    "testTimeout": 60000,
    "transform": {
      "^.+\\.ts$": "ts-jest"
    }
  }
}
```

Comment (ryabukhinstep) [reply packages/app/package.json]: this is a good argument.
fixed

File: packages/app/package.json (hunk 1 above)
Comment (ryabukhinstep) [reply packages/app/package.json]: fixed

**Review by lalumiere17 — DISMISSED** [review]

**Review by KillWolfVlad — APPROVED** [review]

**Pull request merged**